src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/ConsulLongPollCallback.java

```diff
@@ -185,21 +195,27 @@ private boolean contentChanged(byte[] newContent) {
         return !Arrays.equals(oldContent, newContent);
     }
 
-    private long updateIndex(Response response) {
+    private UpdateIndexStatus updateIndex(Response response) {
         String indexString = response.header("X-Consul-Index");
         if (indexString == null) {
             logger.error("There was no X-Consul-Index header in response for {} endpoint, retrying", endpoint);
-            return -1;
+            return UpdateIndexStatus.ERROR;
         }
 
         long index = Long.valueOf(indexString);
-        if (currentIndex.get() == index) {
+        long lastIndex = currentIndex.get();
+
+        if (index == lastIndex) {
             stats.indexNotChanged();
-            logger.trace("Discarding event on endpoint {} index {} as index did not change", endpoint, index);
```

Comment:
is that log removal intentional?

Reply:
yes, the same trace is being logged in `handleContentUnchanged`